### DISCUSSION TITLE
Amino acid order for lm_logits (ESMFold)

### DISCUSSION THREAD

**Discussion by wlin16** [facebookresearch/esm#510] [ESM Metagenomic Atlas]
Hi,

I need to get the ['lm_logits'] from the output of ESMFold to predict which amino acid is most likely at the masked position. However, the dimension of the corresponding ['lm_logits'] for the amino acid is 23. I checked the source code and found that the 23 dimensions consist of pad_idx, 20 amino acids (residue_constants.restype_num (int, 20) from openfold), unk_idx, and mask_idx. 

I want to confirm whether the 20 amino acids in the middle correspond to the amino acid dictionary order of the esm (proteinseq_toks from esm.constants) or the amino acid order of openfold residue_constants.restypes_with_x.

Thank you for your help!

**Comment by ebetica** [comment]
Hi, this is actually legacy code, we never trained ESMfold with an MLM loss. I recommend using esm2 directly.

**Reply by wlin16** [reply]
Thank you for your reply!

I'd like to confirm with you, has ESM-2 learned the structural features of proteins? In my understanding, ESM-2 seems to have learned more about the sequence information of proteins, such as conserved evolutionary information.

Many thanks!
